pkg/api/util/names.go

```diff
@@ -47,6 +46,30 @@ func ComputeName(prefix string, obj interface{}) (string, error) {
 	return fmt.Sprintf("%s-%d", prefix, hashF.Sum32()), nil
 }
 
+// ComputeUniqueDeterministicNameFromData returns a short unique name for the given object.
+func ComputeUniqueDeterministicNameFromData(fullName string) (string, error) {
+	const maxNameLength = 61 // 52 + 1 + 8
+
+	if len(fullName) <= maxNameLength {
+		return fullName, nil
+	}
+
+	hashF := fnv.New32()
```

Comment:
Interesting, maybe we should also increase the limit name length, so there are less characters left that can be used to create collisions?

Reply:
@SgtCoDFish I updated the code, now we embed a full sha256 hash in case the certificate name is longer than 233 characters. (I encoded the hash using hex, just because it is easier & I think we do have more than enough characters available).